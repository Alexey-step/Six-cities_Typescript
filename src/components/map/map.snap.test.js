import React from 'react';
import {render} from '@testing-library/react';
import Map from './map';
import * as redux from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import {offers} from '../../mocks/tests-mocks';

const mockStore = configureStore({});
describe(`Test Map component`, () => {
  jest.spyOn(redux, `useSelector`);
  jest.spyOn(redux, `useDispatch`);
  const history = createMemoryHistory();
  it(`Map component should render correctly at MainPage`, () => {
    const {container} = render(
        <redux.Provider store={mockStore({})}>
          <Router history={history}>
            <Map
              activeLocation={offers.adapted[0]}
              offers={offers.adapted}
              activeCard={offers.adapted[0]}
              mapStyle="MAIN"
            />
          </Router>
        </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
  it(`Map component should render correctly at PropertyPage`, () => {
    const {container} = render(
        <redux.Provider store={mockStore({})}>
          <Router history={history}>
            <Map
              activeLocation={offers.adapted[0]}
              offers={offers.adapted}
              activeCard={offers.adapted[0]}
              mapStyle="PROPERTY"
            />
          </Router>
        </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
