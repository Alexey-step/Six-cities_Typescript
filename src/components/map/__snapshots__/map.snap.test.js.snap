// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Map component Map component should render correctly at MainPage 1`] = `
<div>
  <div
    class="leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="width: 100%; height: 100%; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
              style="width: 256px; height: 256px; left: -130px; top: -72px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 41px; top: -9px; z-index: -9;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 41px; top: -9px; z-index: -9;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin-active.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 41px; top: -9px; z-index: -9;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Map component Map component should render correctly at PropertyPage 1`] = `
<div>
  <div
    class="leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="width: 1144px; height: 579px; margin: 0px auto; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
              style="width: 256px; height: 256px; left: -130px; top: -72px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 41px; top: -9px; z-index: -9;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 41px; top: -9px; z-index: -9;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin-active.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 41px; top: -9px; z-index: -9;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
